internal/datatransfer/monitor.go

```diff
@@ -49,10 +58,69 @@ func (m *Monitor) GetLastSuccessfulSyncTime(workloadName string) *time.Time {
 }
 
 func (m *Monitor) WorkloadRemoved(workloadName string) {
+	m.syncPathsWorkload(workloadName)
+
 	m.lastSuccessfulSyncTimesLock.Lock()
 	defer m.lastSuccessfulSyncTimesLock.Unlock()
 	delete(m.lastSuccessfulSyncTimes, workloadName)
 }
+
+func (m *Monitor) syncPathsWorkload(workloadName string) {
+	storage := m.config.GetDeviceConfiguration().Storage
+	if storage == nil || storage.S3 == nil {
+		return
+	}
+
+	workloads, err := m.workloads.ListWorkloads()
+	if err != nil {
+		log.Errorf("Can't get the list of workloads: %v", err)
+	}
+	if len(workloads) == 0 {
```

Comment:
This check will fail when `WorkloadRemoved` is called for the last workload and the date will not be synchronized for it.
`if len(dataPaths) == 0 {` check few lines is enough to finish execution of the method early and `len(workloads) == 0` check can be removed.